### PR TITLE
feat: Add Storage

### PR DESCRIPTION
**Pull request by FabioPinheiro**
feat: Add Storage

ATL-4581

### CODE REVIEW

File: did-mediator/src/main/scala/fmgp/did/comm/mediator/MediatorAgent.scala [hunk 1]
```diff
@@ -27,19 +33,17 @@ case class MediatorAgent(
   // val resolverLayer: ULayer[DynamicResolver] =
   //   DynamicResolver.resolverLayer(didSocketManager)
 
-  type Services = Resolver & Agent & Operations & MessageDispatcher & Ref[MediatorDB]
-  val protocolHandlerLayer: URLayer[Ref[MediatorDB], ProtocolExecuter[Services]] =
-    ZLayer {
-      for {
-        ref <- ZIO.service[Ref[MediatorDB]]
-      } yield ProtocolExecuterCollection[Services](
+  type Services = Resolver & Agent & Operations & MessageDispatcher & Ref[MediatorDB] & DidAccountRepo & MessageItemRepo
```
Comment (mineme0110) [inline did-mediator/src/main/scala/fmgp/did/comm/mediator/MediatorAgent.scala]: MediatorDB is not required any more

File: did-mediator/src/main/scala/fmgp/did/comm/mediator/MediatorAgent.scala [hunk 2]
```diff
@@ -112,6 +125,9 @@ case class MediatorAgent(
                 *> ZIO.none
             else
               for {
+                messageItemRepo <- ZIO.service[MessageItemRepo]
+                _ <- messageItemRepo.insertOne(MessageItem(msg)) // store all message
+
                 _ <- messageDB.update(db => db.add(msg))
```
Comment (mineme0110) [inline did-mediator/src/main/scala/fmgp/did/comm/mediator/MediatorAgent.scala]: remove this

File: did-mediator/src/main/scala/fmgp/did/comm/mediator/MediatorAgent.scala [hunk 3]
```diff
@@ -137,7 +153,7 @@ case class MediatorAgent(
   def createSocketApp(
       annotationMap: Seq[LogAnnotation]
   ): ZIO[
-    MediatorAgent & Resolver & Operations & MessageDispatcher & Ref[MediatorDB],
+    MediatorAgent & Resolver & Operations & MessageDispatcher & Ref[MediatorDB] & MessageItemRepo & DidAccountRepo,
```
Comment (mineme0110) [inline did-mediator/src/main/scala/fmgp/did/comm/mediator/MediatorAgent.scala]: remove this Ref[MediatorDB]

File: did-mediator/src/main/scala/fmgp/did/comm/mediator/MediatorStandalone.scala [hunk 4]
```diff
@@ -31,11 +32,24 @@ case class MediatorConfig(endpoint: java.net.URI, keyAgreement: OKPPrivateKey, k
...
+  val connectionString = s"$protocol://$userName:$password@$host:$port/$dbName"
+  val displayConnectionString = s"$protocol://$userName:******@$host:$port/$dbName"
+  override def toString: String = s"""DataBaseConfig($protocol, $host, $port, $userName, "******", $dbName)"""
+}
 
 object MediatorStandalone extends ZIOAppDefault {
 
   val app: HttpApp[ // type HttpApp[-R, +Err] = Http[R, Err, Request, Response]
-    Hub[String] & Operations & MessageDispatcher & MediatorAgent & Ref[MediatorDB] & Resolver,
+    Hub[String] & Operations & MessageDispatcher & MediatorAgent & Ref[MediatorDB] & Resolver & MessageItemRepo &
```
Comment (mineme0110) [inline did-mediator/src/main/scala/fmgp/did/comm/mediator/MediatorStandalone.scala]: Ref[MediatorDB] same here remove from siganture

File: did-mediator/src/main/scala/fmgp/did/comm/protocol/ProtocolExecute.scala [hunk 5]
```diff
@@ -84,20 +89,24 @@ trait ProtocolExecuterWithServices[-R <: ProtocolExecuter.Services] extends Prot
...
-                              //   case https if https.startsWith("https://") =>
-                              //     Some(url.drop(8).split(':').head.split('/').head)
-                              //   case _ => None
-                            )
+                            messageDispatcher
+                              .send(
+                                msg,
+                                url, // "http://localhost:8080", // FIXME REMOVE (use for local env)
+                                None
+                                // url match // FIXME REMOVE (use for local env)
```
Comment (mineme0110) [inline did-mediator/src/main/scala/fmgp/did/comm/protocol/ProtocolExecute.scala:101]: Do we need this

File: did-mediator/src/main/scala/fmgp/did/db/MessageItemRepo.scala [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+class MessageItemRepo(reactiveMongoApi: ReactiveMongoApi)(using ec: ExecutionContext) {
+  def collectionName: String = "messages"
+
+  def collection: IO[StorageCollection, BSONCollection] = reactiveMongoApi.database
+    .map(_.collection(collectionName))
+    .mapError(ex => StorageCollection(ex))
+
+  // TODO Rename method
```
Comment (mineme0110) [inline did-mediator/src/main/scala/fmgp/did/db/MessageItemRepo.scala]: fix this insertMessage

**Review by mineme0110 — APPROVED** [review]

File: did-mediator/src/main/scala/fmgp/did/comm/protocol/ProtocolExecute.scala [hunk 7]
```diff
@@ -84,20 +89,24 @@ trait ProtocolExecuterWithServices[-R <: ProtocolExecuter.Services] extends Prot
                           head.getServiceEndpointAsURIs.headOption // TODO head
                         case Seq() => None // TODO
                       }
-                      jobToRun <- mURL match
+                      jobToRun = mURL match
```
Comment (FabioPinheiro) [inline did-mediator/src/main/scala/fmgp/did/comm/protocol/ProtocolExecute.scala:92]: ... merge conflict

**Pull request merged**